hypothesis-python/docs/reference/strategies.rst

```diff
@@ -0,0 +1,230 @@
+.. autofunction:: hypothesis.strategies.decimals
+.. autofunction:: hypothesis.strategies.fractions
+
+Strings
+-------
+
+.. seealso::
+
+  The |st.uuids| and |st.ip_addresses| strategies generate instances of :mod:`UUID <python:uuid>` and :mod:`IPAddress <python:ipaddress>` respectively. You can generate corresponding string values by using |strategy.map|, such as ``st.uuids().map(str)``.
+
+.. autofunction:: hypothesis.strategies.text
+.. autofunction:: hypothesis.strategies.characters
+.. autofunction:: hypothesis.strategies.from_regex
+.. autofunction:: hypothesis.strategies.binary
+.. autofunction:: hypothesis.strategies.emails
+
+Collections
+-----------
+
+.. autofunction:: hypothesis.strategies.lists
+.. autofunction:: hypothesis.strategies.tuples
+.. autofunction:: hypothesis.strategies.sets
+.. autofunction:: hypothesis.strategies.frozensets
+.. autofunction:: hypothesis.strategies.dictionaries
+.. autofunction:: hypothesis.strategies.fixed_dictionaries
+.. autofunction:: hypothesis.strategies.iterables
+
+Datetime
+--------
+
+.. autofunction:: hypothesis.strategies.dates
+.. autofunction:: hypothesis.strategies.times
+.. autofunction:: hypothesis.strategies.datetimes
+.. autofunction:: hypothesis.strategies.timezones
+.. autofunction:: hypothesis.strategies.timezone_keys
+.. autofunction:: hypothesis.strategies.timedeltas
+
+Recursive
+---------
+
+.. autofunction:: hypothesis.strategies.recursive
+.. autofunction:: hypothesis.strategies.deferred
+
+Random
+------
+
+.. autofunction:: hypothesis.strategies.randoms
+.. autofunction:: hypothesis.strategies.random_module
+.. autofunction:: hypothesis.register_random
+
+.. _combinators:
+
+Combinators
+-----------
+
+.. autofunction:: hypothesis.strategies.one_of
+.. autofunction:: hypothesis.strategies.builds
+.. autofunction:: hypothesis.strategies.composite
+.. autofunction:: hypothesis.strategies.data
+
+Typing
+------
+
+.. autofunction:: hypothesis.strategies.from_type
+.. autofunction:: hypothesis.strategies.register_type_strategy
+
+Hypothesis
+----------
+
+.. autofunction:: hypothesis.strategies.runner
+.. autofunction:: hypothesis.strategies.shared
+
+Misc
+----
+
+.. autofunction:: hypothesis.strategies.functions
+.. autofunction:: hypothesis.strategies.slices
+.. autofunction:: hypothesis.strategies.uuids
+.. autofunction:: hypothesis.strategies.ip_addresses
+
+.. autofunction:: hypothesis.strategies.sampled_from
+.. autofunction:: hypothesis.strategies.permutations
+
+Provisional
+-----------
+
+.. automodule:: hypothesis.provisional
+  :members:
+  :exclude-members: DomainNameStrategy
+
+Related
+-------
+
+.. autoclass:: hypothesis.strategies.DataObject
+.. autoclass:: hypothesis.strategies.DrawFn
+
+.. autoclass:: hypothesis.strategies.SearchStrategy
+
+  .. automethod:: hypothesis.strategies.SearchStrategy.filter
+  .. automethod:: hypothesis.strategies.SearchStrategy.map
+  .. automethod:: hypothesis.strategies.SearchStrategy.flatmap
```

Comment:
I think we can use syntax `:members: filter, map, flatmap` for this?

Reply:
this didn't work initially and I gave up trying to figure out why /shrug. It might be because we're overriding `__module__: str = "hypothesis.strategies"` in `SearchStrategy`, which is a pretty cursed thing to be doing in general tbh.